packages/material-ui/src/Button/Button.js

```diff
@@ -387,19 +392,19 @@ Button.propTypes = {
    * The size of the button.
    * `small` is equivalent to the dense button styling.
    */
-  size: PropTypes.oneOf(['small', 'medium', 'large']),
+  size: PropTypes.oneOf(['large', 'medium', 'small']),
   /**
    * Element placed before the children.
    */
   startIcon: PropTypes.node,
   /**
    * @ignore
    */
-  type: PropTypes.string,
+  type: PropTypes.oneOfType([PropTypes.oneOf(['button', 'reset', 'submit']), PropTypes.string]),
```

Comment:
Some redundant checks due to intersecting `button#type` with `a#type`. But we don't document it explicitly anyway so this seems ok-ish.

Reply:
Also: This is actually a fix since it was too narrow when you used the `component` prop e.g. `<Button component="a" type="some/mimetype" />` or `<Button component={MyButtonHandlingArbitraryTypes} type="arbitrary" />` would issue false positive runtime warnings.